topostats/io.py

```diff
@@ -194,25 +212,115 @@ def _ibw_pixel_to_nm_scaling(self, scan) -> float:
         LOGGER.info(f"[{self.filename}] : Pixel to nm scaling : {pixel_to_nm_scaling}")
         return pixel_to_nm_scaling
 
-    def load_jpk(self) -> None:
+    def load_jpk(self) -> tuple:
+        """Loads image from JPK Instruments .jpk files.
+        
+        Returns
+        -------
+        tuple(np.ndarray, float)
+            A tuple containing the image and its pixel to nanometre scaling value.
+        """
+        # Load the file
+        img_path = str(self.img_path)
         try:
-            jpk = load_data(self.img_path)
+            tif = tifffile.TiffFile(img_path)
         except FileNotFoundError:
             LOGGER.info(f"[{self.filename}] File not found : {self.img_path}")
-        except Exception as exception:
-            LOGGER.error(f"[{self.filename}] : {exception}")
-        return jpk
+            raise
+        # Obtain channel list for all channels in file
+        channel_list = {}
+        for i, page in enumerate(tif.pages[1:]):  # [0] is thumbnail
+            available_channel = page.tags["32848"].value  # keys are hexidecimal vals
+            if page.tags["32849"].value == 0:  # wether img is trace or retrace
+                tr_rt = "trace"
+            else:
+                tr_rt = "retrace"
+            channel_list[f"{available_channel}_{tr_rt}"] = i + 1
+        try:
+            channel_idx = channel_list[self.channel]
+        except KeyError:
+            print(f"{self.channel} not in channel list: {channel_list}")
+            raise
+        # Get image and if applicable, scale it
+        channel_page = tif.pages[channel_idx]
+        image = channel_page.asarray()
+        scaling_type = channel_page.tags["33027"].value
+        if scaling_type == "LinearScaling":
+            scaling = channel_page.tags["33028"].value
+            offset = channel_page.tags["33029"].value
+            image = (image * scaling) + offset
+        elif scaling_type == "NullScaling":
+            pass
+        else:
+            raise ValueError(f"Scaling type {scaling_type} is not 'NullScaling' or 'LinearScaling'")
+        # Get page for common metadata between scans
+        metadata_page = tif.pages[0]
+        return (image * 1e9, self._jpk_pixel_to_nm_scaling(metadata_page))
 
-    @staticmethod
-    def _extract_jpk(jpk: AFMCreepCompliance) -> np.ndarray:
-        """Extract data from jpk object"""
+    def _jpk_pixel_to_nm_scaling(self, tiff_page: tifffile.tifffile.TiffPage) -> float:
+        """Extract pixel to nm scaling from the JPK image metadata.
+
+        Parameters
+        ----------
+        tiff_page: tifffile.tifffile.TiffPage
+            An image file directory (IFD) of .jpk files.
+
+        Returns
+        -------
+        float
+            A value corresponding to the real length of a single pixel.
+        """
+        length = tiff_page.tags["32834"].value  # Grid-uLength (fast)
```

Comment:
Will these values _always_ be as stated or is there scope for them changing and needing to be configurable options?

Reply:
Yep this should always be the same according to the jpk specification